core/src/main/java/org/apache/accumulo/core/file/blockfile/impl/CachableBlockFile.java

```diff
@@ -83,8 +84,27 @@ public CachableBuilder conf(Configuration hadoopConf) {
     }
 
     public CachableBuilder fsPath(FileSystem fs, Path dataFile) {
+      return fsPath(fs, dataFile, false);
+    }
+
+    public CachableBuilder fsPath(FileSystem fs, Path dataFile, boolean dropCacheBehind) {
       this.cacheId = pathToCacheId(dataFile);
-      this.inputSupplier = () -> fs.open(dataFile);
+      this.inputSupplier = () -> {
+        FSDataInputStream is = fs.open(dataFile);
+        if (dropCacheBehind) {
+          // Tell the DataNode that the write ahead log does not need to be cached in the OS page
+          // cache
+          try {
+            is.setDropBehind(Boolean.TRUE);
```

Comment:
I was wondering how I could know if this changes was working as intended, thought adding some trace logging that could be flipped on might be useful.

```suggestion
            is.setDropBehind(Boolean.TRUE);
            log.trace("Called setDropBehind(TRUE) for stream reading file {}", dataFile);
```